ui/src/app/pages/signed-in/signed-in-nav-bar.tsx

```diff
@@ -0,0 +1,223 @@
+  hasDataAccess: boolean;
+  headerImg: string;
+  displayTag: string;
+  shouldShowDisplayTag: boolean;
+  givenName: string;
+  familyName: string;
+  aouAccountEmailAddress: string;
+  contactEmailAddress: string;
+  profileImage: string;
+  sidenavToggle: boolean;
+  homeActive: boolean;
+  workspacesActive: boolean;
+  libraryActive: boolean;
+  profileActive: boolean;
+}
+
+export interface State {
+  sideNavVisible: boolean;
+  barsTransform: string;
+  hovering: boolean;
+  wrapperRef: React.RefObject<HTMLDivElement>;
+}
+
+const barsTransformNotRotated = 'rotate(0deg)';
+const barsTransformRotated = 'rotate(90deg)';
+
+export const SignedInNavBar = withUserProfile()(
+  class extends React.Component<Props, State> {
+    constructor(props) {
+      super(props);
+      // Bind the this context - this will be passed down into the actual
+      // sidenav so clicks on it can close the nav
+      this.onToggleSideNav = this.onToggleSideNav.bind(this);
+      this.handleClickOutside = this.handleClickOutside.bind(this);
+      this.state = {
+        sideNavVisible: false,
+        barsTransform: barsTransformNotRotated,
+        hovering: false,
+        wrapperRef: React.createRef(),
+      };
+    }
+
+    onToggleSideNav() {
+      this.setState(previousState => ({sideNavVisible: !previousState.sideNavVisible}));
+      this.setState(previousState => ({
+        barsTransform: previousState.barsTransform === barsTransformNotRotated
+          ? barsTransformRotated
+          : barsTransformNotRotated
+      }));
+    }
+
+    componentDidMount() {
+      document.addEventListener('click', this.handleClickOutside);
+    }
+
+    componentWillUnmount() {
+      document.removeEventListener('click', this.handleClickOutside);
+    }
+
+    handleClickOutside(event) {
+      if (
+        this.state.wrapperRef
+        && !this.state.wrapperRef.current.contains(event.target)
+        && this.state.sideNavVisible
+      ) {
+        this.onToggleSideNav();
+      }
+    }
+
+    render() {
+      return <div
+        style={styles.headerContainer}
+        ref={this.state.wrapperRef}
+      >
+        <div style={{
+          transform: this.state.barsTransform,
+          display: 'inline-block',
+          marginLeft: '1rem',
+          transition: 'transform 0.5s',
+        }}>
+          <ClrIcon
+            shape='bars'
+            onClick={() => this.onToggleSideNav()}
+            onMouseEnter={() => this.setState({hovering: true})}
+            onMouseLeave={() => this.setState({hovering: false})}
+            style={this.state.hovering
+              ? {...styles.sidenavIcon, ...styles.sidenavIconHovering}
+              : {...styles.sidenavIcon}}
+          >
+          </ClrIcon>
+        </div>
+        <div>
+          <a href={'/'}>
+            <img
+              src={this.props.headerImg}
+              style={styles.headerImage}
+            />
+          </a>
+          {
+            this.props.shouldShowDisplayTag
```

Comment:
this could be like {this.props.shouldShowDisplayTag && <div style={styles.displayTag}>...</div>}